### ISSUE TITLE
Task: change different order in the block transforms menu

### ISSUE THREAD

**Issue by simison** [Automattic/wp-calypso#67635]
### Details

Currently “premium content” and “conversation” seem to _always_ appear as the top suggestions in the transforms menu:

![image](https://user-images.githubusercontent.com/87168/189597883-3bf8a6e0-e77d-4600-8c59-6eee3ae16b4d.png)

In many cases, basic core blocks like "columns", "heading", "list",  "paragraph" should be suggested first and less frequently used Premium content (and especially" conversations") should be lower in the list.

On an older Atomic site with Business plan:

<img width="676" alt="Screenshot 2022-09-12 at 10 38 16" src="https://user-images.githubusercontent.com/87168/189598641-cd53d559-6cc6-4bd9-bc11-42c7cdf06f00.png">
<img width="539" alt="Screenshot 2022-09-12 at 10 38 04" src="https://user-images.githubusercontent.com/87168/189598643-2a4d523f-c7b2-4efc-93c2-88ef42fdf6fa.png">
<img width="912" alt="Screenshot 2022-09-12 at 10 37 54" src="https://user-images.githubusercontent.com/87168/189598645-b8036647-fabe-4c7b-b13b-5686334ade1a.png">

In above screenshots some other CoBlocks plugin blocks also appear on top, so this could be another core bug like we had before with global styles lists.

### Related

https://github.com/WordPress/gutenberg/issues/40208

**Comment by annezazu** [comment]
Opened an issue about this today -- In Core, we're auditing block transforms https://github.com/WordPress/gutenberg/issues/63635 and I wanted to note how prominent and confusing the paid content transform on WordPress.com is as part of looking across our products. Here's an example from the blog home template on my personal nomad.blog site:


https://github.com/user-attachments/assets/94114c69-61ce-49e3-9262-4a583962bf4f


I'm curious how effective this is, especially if we have any data, and whether we can make it more specific. For example, it shouldn't appear for a site title or menu but I can understand why it might show up for group blocks. Either way, it's currently taking up precious real estate visually, despite not being a common use case.

**Comment by simison** [comment]
Yeah; leaving the transform appear in top 3 just for container blocks makes sense, and can be lower/below the core blocks for the other blocks.

**Comment by simison** [comment]
@obenland is there a .com team who could tweak the transform priorities on WP.com? See Anne's feedback above.

This could be pretty straightforward.

**Comment by simison** [comment]
Picked this for Zap ⚡ 

Plan is to see if we can keep showing "convert to paid blocks" only when selecting multiple blocks, OR when not selecting group/columns blocks hides it from menus elsewhere.

Otherwise we can limit to just group/columns blocks using the `blocks` attribute in transform definitions.

Ideally we'd just push the paid transform down on the list below core blocks, but that might not be possible now in Gutenberg.

<img width="728" alt="Screenshot 2024-07-26 at 18 01 40" src="https://github.com/user-attachments/assets/1d16b65f-cde3-45d6-8d33-8eee71b5966c">

**Comment by lezama** [comment]
I looked into this a bit. Transformations are listed in the order that blocks.registerBlockType calls are made, with an additional orderBy for frecency. It appears that core/blocks are registered after plugin blocks. Generally, I think it would make more sense to register core blocks before plugin blocks.

**Comment by simison** [comment]
> It appears that core/blocks are registered after plugin blocks. Generally, I think it would make more sense to register core blocks before plugin blocks.

Any existing Gutenberg issues about this or wanna write a suggestion?
